plugins/org.jboss.tools.vpe.preview.core/src/org/jboss/tools/vpe/preview/core/util/ActionBar.java

```diff
@@ -219,7 +219,9 @@ public boolean isAutomaticRefreshEnabled() {
 	
 	public void updateRefreshItemsAccordingToPreferences() {
 		enableRefreshOnSaveAction.setChecked(preferences.getBoolean(REFRESH_ON_SAVE_PREFERENCES));
+		enableRefreshOnSaveAction.run();
 		enableAutomaticRefreshAction.setChecked(preferences.getBoolean(REFRESH_ON_CHANGE_PREFERENCES));
+		enableAutomaticRefreshAction.run();
```

Comment:
yes, thats correct. Setting action as checked wasn't enough because listeners were removed before. Another part of incorrect test was that listeners listen to save command not editor.save(). 

It wasn't discovered before because it seems to be that updateRefreshItemsAccordingToPreferences method was added just to be able to test these actions.